ccl/partitionccl: TestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill failed

ccl/partitionccl.TestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill [failed](https://mesolite.cluster.engflow.com/invocations/default/26ee6ef0-5910-46ae-a691-e2114712c4a3?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3BhcnRpdGlvbmNjbDpwYXJ0aXRpb25jY2xfdGVzdA==) on master @ [c8c2809e2506a6cbadb1cb93dbc46383101917f0](https://github.com/cockroachdb/cockroach/commits/c8c2809e2506a6cbadb1cb93dbc46383101917f0):


```
=== RUN   TestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill1817061103
    test_log_scope.go:82: use -show-logs to present logs inline
    alter_primary_key_test.go:48: -- test log scope end --
test logs left over in: outputs.zip/logTestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill1817061103
--- FAIL: TestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill (475.66s)
=== RUN   TestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill/ALTER_PRIMARY_KEY
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    alter_primary_key.go:121: 
        	Error Trace:	pkg/ccl/testutilsccl/alter_primary_key.go:121
        	Error:      	Received unexpected error:
        	            	dial tcp 127.0.0.1:33395: connect: connection refused
        	Test:       	TestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill/ALTER_PRIMARY_KEY
--- FAIL: TestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill/ALTER_PRIMARY_KEY (475.65s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54715

--- a/pkg/ccl/partitionccl/BUILD.bazel
+++ b/pkg/ccl/partitionccl/BUILD.bazel
@@ -77,6 +77,7 @@ go_test(
         "//pkg/sql/types",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util/encoding",

--- a/pkg/ccl/partitionccl/alter_primary_key_test.go
+++ b/pkg/ccl/partitionccl/alter_primary_key_test.go
@@ -9,6 +9,7 @@ import (
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/ccl/testutilsccl"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 )
@@ -16,6 +17,9 @@ import (
 func TestAlterPrimaryKeyCorrectZoneConfigBeforeBackfill(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	skip.UnderDeadlock(t)
+	skip.UnderRace(t)
 
 	testCases := []testutilsccl.AlterPrimaryKeyCorrectZoneConfigTestCase{
 		{

